clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -50,6 +50,7 @@
 
     private final long refreshBackoffMs;
     private final long metadataExpireMs;
+    private final long metadataStaleMs;
```

Comment:
As discussed offline, incorporated `metadataExpireMs` and `refreshBackOffMs` into consideration. Updated `muted` partitions back into `abortExpiredBatches`.